.pipelines/.vsts-vhd-builder.yaml

```diff
@@ -104,3 +104,18 @@ stages:
       LOCATION: $(PACKER_BUILD_LOCATION)
     jobs:
       - template: ./templates/e2e-template.yaml
+  - stage: Generate and Publish Testdata
+    condition: succeeded()
+    jobs:
+      - job: Generate_Publish_Testdata
+        timeoutInMinutes: 10
+        steps:
+        - bash: |
+            make generate
+          displayName: Generate Testdata
+        - task: 1ES.PublishPipelineArtifact@1
+          inputs:
+            targetPath: $(Pipeline.Workspace)/pkg/agent/testdata
```

Comment:
how would we consume this from unit test workflows? iterate through the most recent runs and pick out the latest one from master?

I was also originally thinking we'd have a new workflow that would only trigger on merge commits to the master branch which would actually upload/update the testdata, as this stands we'd upload testdata as an artifact to every VHD check-in build

Reply:
it wont be a workflow, I'm working on a new ado pipeline.

Reply:
what are checkins build ?

Reply:
VHD check-in builds, which is defined by `.vsts-vhd-builder.yaml`

Reply:
I"m not sure where else we should upload the testdata ? why not on the master build after things gets merged ?

Reply:
I think builds off the master branch are fine to upload from, though with this change we'd be uploading artifacts to every instance of the checkin build

Reply:
That's also why I was wondering how we'd pick out the correct artifacts from the new pipeline

Reply:
Ah okay, I guess V2 of DownloadPipelineArtifact will default to latest from master branch: https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/download-pipeline-artifact-v2?view=azure-pipelines

Reply:
correct, you can also select the version. but I was thinking of pulling latest from Master all the time